include/boost/gil/point.hpp

```diff
@@ -71,9 +72,20 @@ class point
         return *this;
     }
 
-    point& operator/=(double t)
+    point& operator/=(double d)
     {
-        if (t < 0 || 0 < t) { x /= t; y /= t; }
+        if (d < 0 || 0 < d)
+        {
+            x = static_cast<T>(x / d);
+            y = static_cast<T>(y / d);
+        }
```

Comment:
Can you elaborate on how these explicit casts differ from the previous implicit ones in "x /= t" ?

Reply:
Both conversions are equivalent. The implicit just rubs the compiler, especially in strict warnings mode, namely:

```
warning C4244: '/=': conversion from 'double' to 'T', possible loss of data

warning: conversion from ‘double’ to ‘int’ may change value [-Wfloat-conversion]
             x /= d;
             ~~^~~~
```

 The explicit states the intention, well, explicit and helps to avoid the warnings.